tests/test_producers.py

```diff
@@ -214,3 +214,21 @@ def test_parameter_validation(self):
 
             with self.assertRaises(ValueError):
                 mbq.atomiq.celery_publish(mock.Mock())
+
+    def test_synchronous_execution_with_setting(self):
+        self.mock_task.name = str(uuid.uuid4())
+        args = ('there', 'is', 'no', 'spoon')
+        kwargs = {
+            'one': 'fish',
+            'two': 'fish',
+            'red': False,
+            'blue': 42,
+        }
+
+        with self.settings(ATOMIQ={'synchronous': True}):
+            with transaction.atomic():
+                mbq.atomiq.celery_publish(self.mock_task, *args, **kwargs)
+
+        models.CeleryTask.objects.get(task_name=self.mock_task.name)
+
+        self.mock_task.assert_called_once_with(*args, **kwargs)
```

Comment:
Should we also test that the task _wasn't_ called when "synchronous" is not passed in? Only reason I say that is it would be very bad if tasks run synchronously if they are not supposed to.